crates/sdk/src/signing.rs

```diff
@@ -405,45 +461,101 @@ pub async fn aux_signing_data(
             Some(AccountPublicKeysMap::from_iter(public_keys.clone())),
             1u8,
         ),
-        Some(owner @ Address::Internal(internal)) => match internal {
-            InternalAddress::Masp => (None, 0u8),
-            _ => {
+        Some(owner @ Address::Internal(_)) => {
+            return Err(Error::from(TxSubmitError::InvalidAccount(
+                owner.encode(),
+            )));
+        }
+        None => (
+            Some(AccountPublicKeysMap::from_iter(public_keys.clone())),
+            0u8,
+        ),
+    };
+    let fee_auth = match wrap_tx {
+        args::Wrapper {
+            wrapper_fee_payer: Some(pubkey),
+            ..
+        } => FeeAuthorization::Signer {
+            pubkey: pubkey.to_owned(),
+            disposable_fee_payer: false,
+        },
+        _ => {
+            if let Some(pubkey) = public_keys.first() {
+                FeeAuthorization::Signer {
+                    pubkey: pubkey.to_owned(),
+                    disposable_fee_payer: false,
+                }
+                // NOTE: the disposable gas_payer is always overridden by
+                // signing inputs
+            } else if disposable_gas_payer {
```

Comment:
Technically if a disposable gas payer is required no signer should be provided since it's not needed, grows the size of the tx (and the gas required) and might leak information. Still, I've maintained the same logic we've had in place for a while for the wrapper signer: if the user specifically passes a signing key we still produce a signature for that (i.e. the user is still in control even though we try to guide them)